### DISCUSSION TITLE
How can I test production mode without errors, after having used development mode?

### DISCUSSION THREAD

**Discussion by pleirb** [Tecnativa/doodba-copier-template#387] [Q&A] [answered]
I've noticed that most of the invoke commands seem to be useful for development mode, don't they? Somebody correct me if I'm wrong. To use in production mode I am trying with:
`docker-compose -f prod.yaml up -d`
But when I view with docker ps I can see that there is no port exposed as in development mode. So I draw the conclusion that it must be that I need to start Traefik as explained in:https://github.com/Tecnativa/doodba-copier-template/blob/main/docs/faq.md#how-to-bootstrap-the-global-inverse-proxy
I create the inverseproxy.yam file with what is suggested for Traefik 1 (because 2 is not available, I don't know why) and run the command:
`docker-compose -p inverseproxy -f inverseproxy.yaml up -d`
To test, since I don't have a domain yet, I add a test line in the hosts file
127.0.0.1 test.com
But when I go to try the browser I get a blank page that says: "Bad Gateway"
I have been trying to understand how Traefik works to understand what is happening to me. I understand that this service is an autorouter that reads the running containers looking for labels that tell it what to do with that container and how to associate it to the corresponding domain, for example. But I see a lot of tags generated by copier and I have a hard time understanding it. On the other hand, I see that Traefik is in newer versions, and in the documentation the example for version 2 seems not to be available. It also surprises me that the Traefik management port in the Traefik 1 example was not exposed, is there a reason for this? Sorry for the number of inquiries, just by clearing some of them I will be very happy. Thank you!

**Comment by carlosecv** [comment]
It’s correct on productive mode you must use docker-compose commands

**Reply by pleirb** [reply]
> 

Yes, I have noticed that the invoke commands sometime rewrite the symbolic link to devel.yaml. For which it is advisable to use the invoke commands only for development, right?

**Comment by yajo** [comment]
https://sslip.io/

**Reply by pleirb** [reply]
What a difference there is in using this and the host file to test domains in production mode. I would like to understand. Is it because of the security certificates? Thanks for the answers!

**Reply by pleirb** [reply, marked as answer]
To solve, what they comment here has helped me: https://github.com/Tecnativa/doodba/issues/418
